backend/hdf5/h5x/H5DataType.cpp

```diff
@@ -9,17 +9,14 @@
 
 #include "H5DataType.hpp"
 
-#include <stdexcept>
-#include <iostream>
-#include <cassert>
-#include <string>
 #include <memory>
-#include <cstdlib>
+#include <malloc.h>
 
```

Comment:
Do we need the malloc include? Under linux and OS X it works without? If windows needs it, I am afraid that we need to find a way to only include it if os == win